cartscript.js

```diff
@@ -0,0 +1,77 @@
+
+const cart=document.querySelector("cart")
+const cartRemove_btns= document.querySelectorAll(".trash-btn");
+addCartRemoveEvent();
+changingItemQuantityEvent();
+itemLikeEvent();
+
+///////// deleting item from a cart
+function addCartRemoveEvent(){
+    cartRemove_btns.forEach( function(btn) {
+        btn.addEventListener("click", handleRemoveCart);
+    })
+}
+
+///// changing quantity of an item in the cart
+function changingItemQuantityEvent(){
+    let cartQuantities= document.querySelectorAll(".cart-quantity");
+    cartQuantities.forEach(function(cartQuantity){
+        cartQuantity.addEventListener("change" , handle_ChangeItemQuantity);
+    })
+}
+
+//// clicking like button or heart Icon
+function itemLikeEvent(){
+    let heartBtns = document.querySelectorAll(".like-btn");
+    heartBtns.forEach( function(heartBtn){
+       // console.log(heartBtn.value);
+        heartBtn.addEventListener("click", handle_LikeItem);
+    })
+}
+function handle_LikeItem(){
+    let c=this.className;
+    console.log(c);
+    if( c == "fa fa-heart like-btn notLiked"){
+         this.style.color= "red";
+         this.className="fa fa-heart like-btn Liked";
+    }
+    else{
+        this.style.color= "white";
+        this.className="fa fa-heart like-btn notLiked";
+    }
+    
+  
+    
+}
+function handle_ChangeItemQuantity(){
+  if(this.value < 1 ){
+    this.value=1;
+  }
+  updateCartTotal();
+}
+
+ function handleRemoveCart(){
+    console.log("handling cart deletion");
+    let parent=this.parentElement.parentElement;
+    console.log(parent);
+    parent.remove();
+    updateCartTotal();
+ }
+
+function updateCartTotal(){
+    let cartItems= document.querySelectorAll(".cart-item");
+    const totalCartPrice= document.querySelector(".total-price");
+    let total=0;
+    cartItems.forEach( function(cartItem){
+        let priceElement= cartItem.querySelector(".cart-price");
+        ////converting string into a number & removing dollar sign to do our calculatipns
```

Comment:
I think it's better to use a custom attribute in the HTML and put in it the price of each item rather than depending on the text displayed to the user in the HTML because it's not a very reliable source.

This can be achieved in the HTML by using the custom `data` attribute like this:

```html
<div class="cart-price" data-price="30">$30.00</div>
```

and access this like in your JS code:

```js
const priceElement = cartItem.querySelector('[data-price]');
const price = Number(priceElement.dataset.price);
```